### ISSUE TITLE
Accuracy oscillates between ~0% and ~70% when creating multiple models

### ISSUE THREAD

**Issue by ChrisSwinchatt** [tensorflow/tensorflow#22033]
### System information
- **Have I written custom code (as opposed to using a stock example script provided in TensorFlow)**: yes
- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**: Windows 10
- **Mobile device (e.g. iPhone 8, Pixel 2, Samsung Galaxy) if the issue happens on mobile device**: No
- **TensorFlow installed from (source or binary)**: PyPi (pip)
- **TensorFlow version (use command below)**: b'v1.10.0-rc1-19-g656e7a2b34' 1.10.0
- **Python version**: 3.6.6
- **Bazel version (if compiling from source)**: N/A
- **GCC/Compiler version (if compiling from source)**: N/A
- **CUDA/cuDNN version**: CUDA 9.0
- **GPU model and memory**: GTX 1060 6 GB
- **Exact command to reproduce**:
  - Download bug.py and training.csv from this Gist: https://gist.github.com/ChrisSwinchatt/97304761e9f875dfd34e3339891a5475
  - Run python bug.py

### Describe the problem

**I've found that the problem doesn't occur when TensorFlow is forced to use the CPU**

Full disclosure: I also opened an issue with Keras (https://github.com/keras-team/keras/issues/11070) because I'm not sure whether the bug is with TensorFlow or Keras. I am using tf.Keras, though, rather than just Keras with TensorFlow as a backend.

I'm using a sequence-to-sequence model based on the Keras blogpost which I've wrapped into a fairly complicated object (although the issue also occurs with a simplified version linked below). When I create a new model (which I have to do for gridsearch and for clearing the TF session when the graph gets too big and slows down training) it starts with accuracy of either 0% or 70%.

Here are a pair of screenshots that show what I mean:

Good: https://i.imgur.com/7mT5Siv.png
Bad: https://i.imgur.com/MZ3NdCB.png

You can see in the first screenshot that the accuracy is low but trending upwards. In the second, the accuracy of two models starts at 70% and doesn't increase (another model starts at 3% and also doesn't increase).

This happens whether I create new, blank models or load pretrained weights into new models with model.load_weights().

### Source code / logs

Source code and screenshots are above.

**Comment by omalleyt12** [comment]
@ChrisSwinchatt could you please share a simple code snippet that reproduces this issue?

**Comment by bignamehyp** [comment]
This question is better asked on [StackOverflow](http://stackoverflow.com/questions/tagged/tensorflow) since it is not a bug or feature request. There is also a larger community that reads questions there.

If you think we've misinterpreted a bug, please comment again with a clear explanation, as well as all of the information requested in the [issue template](https://github.com/tensorflow/tensorflow/issues/new). Thanks!

**Comment by ChrisSwinchatt** [comment]
@omalleyt12 

I did, please take a look at my Gist: https://gist.github.com/ChrisSwinchatt/97304761e9f875dfd34e3339891a5475

The bug is not triggered if you add:
```python
    with tf.Session(config=tf.ConfigProto(device_count={'GPU':0})) as session:
        K.set_session(session)
        # ...
```

@bignamehyp 
I think I did include all the (relevant) information in the issue template. I'm not in need of support because I have found that training on CPU alleviates the problem. I just think it's something you should be aware of in case there is a bug in TensorFlow. Of course, it's possible it's actually a bug in CUDA, CUDNN, the drivers, Windows or even my GPU (purchased recently).

**The bug is that when training on GPU and creating a second (or third, or fourth, or *N*th) model, its accuracy on the first batch/epoch will be either ~0% or ~70% and never really improve.** The first model works as expected but this obviously isn't good for model selection/cross validation where we're obliged to make several models.

Please let me know if any other information will help you.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @bignamehyp: It has been 14 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @bignamehyp: It has been 29 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by wt-huang** [comment]
@ChrisSwinchatt This problem is likely caused by your GPU setup. Try to do a sanity check first and use a small sample first.

**Comment by wt-huang** [comment]
Closing this as it is not a bug nor feature request, feel free to open a new issue if problem persists.
